AngularJS/src/app.run.js

```diff
@@ -0,0 +1,12 @@
+angular.module('app-bootstrap').run([
+'$transitions', 'sessionService',
+  function ($transitions, sessionService) {
+    $transitions.onBefore({}, function(transition) {
+      if (transition.to().data.requireLogin && !sessionService.loginState()) {
+        return transition.router.stateService.target('registration');
+      } else if (!transition.to().data.requireLogin && sessionService.loginState()) {
```

Comment:
This can be:

```js
if (transition.to().data.requireLogin) {
  if (sessionService.loginState()) {
    return transition.router.stateService.target('navbar.home');
  }
  return transition.router.stateService.target('registration');
}
```

But I think you are also missing a default return 🤔